### DISCUSSION TITLE
Create component with REST API for an additional branch

### DISCUSSION THREAD

**Discussion by mattheop** [WeblateOrg/weblate#9329] [Q&A]
Hey, weblate offers a great "Create a new translation component for an additional branch of the existing translation component" feature when adding a component. I'd like to know if it's possible to trigger this creation behavior with the REST API?

**Comment by nijel** [comment]
It just copies settings and creates a new component with a different branch. With API, GET existing component, change name, slug and branch. And POST to create new one.

**Reply by mattheop** [reply]
This is what I wanted to do, but the extensions are not copied. if I POST the addon property from the GET, addon are not added to the new one.

Example:
Here is my POST payload
```json
{
    "file_format": "json",
    "name": "my name",
    "slug": "my-slug",
    "branch": "new-feat",
    "push_branch": "new-feat",
    "filemask": "app/locales/*.json",
    "template": "app/locales/en.json",
    "repo": "git@gitlab.com:XXXXX.XXXXX/XXXXX.git",
    "allow_translation_propagation": false,
    "addons": [
	"https://weblate.dev/api/addons/7"
     ]
}
```
`addons` is exactly what i get from GET request of the "parent" one.
But the extension is not added to the new one, if i make a GET on the created component, `addons` is an empty array
More over, if i try to manually replace the addon URL by the name, it doesn't work either. And according to the documentation, there is no addon property ([see](https://docs.weblate.org/en/latest/api.html#get--api-components-(string-project)-(string-component)-))

Maybe I'm missing something ?

**Reply by nijel** [reply]
The UI option IMHO does not clone add-ons as well. You need to install these manually in both cases.
